### PR TITLE
For #1313: Additional ctors for `Filtered`

### PR DESCRIPTION
**Pull request by andreoss**
This is for #1313: 

- Add ctors with `Func<X, Scalar<Boolean>>` to `Filtered`

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#1402](https://github.com/yegor256/cactoos/issues/1402) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/6f4521a0-b20b-4115-a089-9c0c4266109b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/org/cactoos/scalar/Reduced.java [hunk 1]
```diff
@@ -97,6 +97,21 @@ public Reduced(
         this.function = reduce;
     }
 
+    /**
+     * Ctor.
+     * @param values The scalars
+     * @param reduce Reducing function
+     */
+    public Reduced(
```
Comment (victornoel) [inline src/main/java/org/cactoos/scalar/Reduced.java:105]: @andreoss if you introduce a new secondary constructor, put it before the primary, not after

File: src/main/java/org/cactoos/text/EndsWith.java [hunk 2]
```diff
@@ -62,6 +62,15 @@ public EndsWith(final Text origin, final Text suffix) {
         this.suffix = suffix;
     }
 
+    /**
+     * Ctor.
+     * @param origin The origin
+     * @param suffix The suffix
+     */
+    public EndsWith(final String origin, final String suffix) {
```
Comment (victornoel) [inline src/main/java/org/cactoos/text/EndsWith.java:70]: @andreoss when you introduce a secondary constructor, please put it before the primary one, not after.

File: src/test/java/org/cactoos/iterable/FilteredTest.java [hunk 3]
```diff
@@ -130,12 +138,63 @@ public void withItemsNotEmpty() {
...
+
+    @Test
+    public void filtersWithTwoFuncsCombined() throws Exception {
+        new Assertion<>(
+            "Must be filtered with two filters",
+            new Filtered<>(
+                new IterableOf<>("ay", "xb", "yx", "xy"),
+                FilteredTest.combine(
+                    input -> new StartsWith(input, "x"),
+                    input -> new EndsWith(input, "y")
```
Comment (victornoel) [inline src/test/java/org/cactoos/iterable/FilteredTest.java:169]: @andreoss what is test really testing? Clearly not `Filtered` since it is tested similarly in the previous method

Comment (andreoss) [reply src/test/java/org/cactoos/iterable/FilteredTest.java:169]: I was trying to emphasize the need for `func.And`, so I've added this test and the puzzle. 
I will remove the test above test than.

**Comment by andreoss** [comment]
> @andreoss see my comments

@victornoel Done

File: src/test/java/org/cactoos/iterable/FilteredTest.java (hunk 3 above)
Comment (victornoel) [reply src/test/java/org/cactoos/iterable/FilteredTest.java:157]: @andreoss I think we should keep the other one that you deleted and remove all the extra constructors you added that are not needed and also to remove this combine method.The other test 1) was simpler 2) had less dependencies 3) perfectly did the job of testing `Filtered`.

File: src/test/java/org/cactoos/iterable/FilteredTest.java [hunk 4]
```diff
@@ -130,12 +133,24 @@ public void withItemsNotEmpty() {
...
+            new Filtered<>(
                 input -> input.length() > 16,
                 new IterableOf<>("third", "fourth")
             ),
             new IsEmptyIterable<>()
         ).affirm();
     }
 
+    @Test
+    public void filtersWithTwoFuncsCombined() throws Exception {
```
Comment (victornoel) [inline src/test/java/org/cactoos/iterable/FilteredTest.java:145]: @andreoss I believe the name of the test is incorrect?

File: src/main/java/org/cactoos/scalar/Reduced.java [hunk 5]
```diff
@@ -84,6 +84,21 @@
      */
     private final BiFunc<T, T, T> function;
 
+    /**
+     * Ctor.
+     * @param values The scalars
+     * @param reduce Reducing function
+     */
+    public Reduced(
```
Comment (victornoel) [inline src/main/java/org/cactoos/scalar/Reduced.java:92]: @andreoss I blieve this is out of scope of this PR now, no?

File: src/main/java/org/cactoos/text/EndsWith.java [hunk 6]
```diff
@@ -43,6 +43,15 @@
      */
     private final Text suffix;
 
+    /**
+     * Ctor.
+     * @param origin The origin
+     * @param suffix The suffix
+     */
+    public EndsWith(final String origin, final String suffix) {
```
Comment (victornoel) [inline src/main/java/org/cactoos/text/EndsWith.java:51]: @andreoss this is also out of scope of this PR.

**Comment by victornoel** [comment]
@andreoss by the way, if you don't notify the REV that you are finishing with fixing the remarks, REV doesn't know about it. Luckily I went to see what was happening here but I could have missed it. It's good practice to simply ping the REV once you fixed everything :)

File: src/main/java/org/cactoos/scalar/Reduced.java (hunk 5 above)
Comment (andreoss) [reply src/main/java/org/cactoos/scalar/Reduced.java]: Removed

File: src/main/java/org/cactoos/text/EndsWith.java (hunk 6 above)
Comment (andreoss) [reply src/main/java/org/cactoos/text/EndsWith.java]: Added it for a test, which got removed. Done

File: src/test/java/org/cactoos/iterable/FilteredTest.java (hunk 4 above)
Comment (andreoss) [reply src/test/java/org/cactoos/iterable/FilteredTest.java]: @victornoel Done

**Comment by andreoss** [comment]
@victornoel Do you mean GH notifications or something else?

**Comment by victornoel** [comment]
@andreoss I mean just saying "heh @victornoel, I made the change, please take a look again". Thanks for the reverts.

@paulodamaso it should be good now

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
Code review was too long (10 days), architects (@paulodamaso, @victornoel[/z](https://www.0crat.com/u/victornoel)) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)
